spring-integration-core/src/main/java/org/springframework/integration/channel/interceptor/ThreadStatePropagationChannelInterceptor.java

```diff
@@ -113,6 +117,20 @@ public MessageHeaders getHeaders() {
 		}
 
 		@Override
+		public MessageBuilder<Object> cloneMessage() {
+			MessageBuilder<Object> messageBuilder = MessageBuilder.fromMessage(this.message);
```

Comment:
Eh... The problem with `sequence` and `MutableMessage` that all the subscribers will receive the same message with identical `sequenceNumber`. That's why `MutableMessageBuilder` is fully wrong from `PublishSubscribeChannel`.

That's one.

Here we return `MessageBuilder` anyway. Nobody expect any other abstraction because of `CloneableMessage` contract. So, I'm not sure that it sounds reasonable to "mutate" something here if that doesn't fulfill for contract.

But, if you insist, I can reinstate factory here easily.